match_handler.go

```diff
@@ -121,12 +126,20 @@ func (m *MatchHandler) MatchInit(ctx context.Context, logger runtime.Logger, db
 		labelJSON = []byte("{}")
 	}
 
-	return &MatchState{
-		random: rand.New(rand.NewSource(time.Now().UnixNano())),
-		label:  label,
-
+	state := &MatchState{
+		random:    rand.New(rand.NewSource(time.Now().UnixNano())),
+		label:     label,
+		ai:        ai,
 		presences: make(map[string]runtime.Presence, 2),
-	}, tickRate, string(labelJSON)
+		messages:  make(chan runtime.MatchData, 1),
+	}
+
+	// Automatically add AI player
+	if ai {
+		state.presences[aiUserId] = aiPresenceObj
```

Comment:
If both players leave, will the match shut down or will the ai presence prevent this from happening?

Reply:
In the case of an AI match there will be just one human player, but good catch, I'll double check if it properly shuts down.